src/common/file_system.cpp

```diff
@@ -314,6 +314,10 @@ FileType FileSystem::GetFileType(FileHandle &handle) {
 	return FileType::FILE_TYPE_INVALID;
 }
 
+FileType FileSystem::TryGetFileType(const std::string &path, optional_ptr<string> error) {
```

Comment:
If the `error` is meant to return the error if it occurs, this should be a `optional_ptr<string> &error`
I'm a little confused as to why this is an `optional_ptr` and not just a `string &` though.

Reply:
I ended up going with a string* so I could set a nullptr default - thoughts on that approach?

Reply:
That's exactly what we have `optional_ptr` for, I still think it should be a `string &error`

Reply:
Ok sounds good I ended up going with optional_ptr to allow for the default. I add a null check for the optional_ptr to see if it's valid. If it is - I will modify the passed in string